peers/app_request_network.go

```diff
@@ -126,95 +133,155 @@ func NewNetwork(
 		return nil, err
 	}
 
-	validatorClient := validators.NewCanonicalValidatorClient(logger, cfg.GetPChainAPI())
+	for _, peer := range manuallyTrackedPeers {
+		logger.Info(
+			"Manually Tracking peer (startup)",
+			zap.String("ID", peer.ID.String()),
+			zap.String("IP", peer.PublicIP.String()),
+		)
+		testNetwork.ManuallyTrack(peer.ID, peer.PublicIP)
+	}
 
-	arNetwork := &appRequestNetwork{
-		network:              testNetwork,
-		handler:              handler,
-		infoAPI:              infoAPI,
-		logger:               logger,
-		lock:                 new(sync.Mutex),
-		validatorClient:      validatorClient,
-		metrics:              metrics,
-		manuallyTrackedPeers: manuallyTrackedPeers,
+	// Connect to a sample of the primary network validators
+	peers, err := infoAPI.Peers(context.Background(), nil)
+	if err != nil {
+		logger.Error(
+			"Failed to get peers",
+			zap.Error(err),
+		)
+		return nil, err
 	}
+	pClient := platformvm.NewClient(cfg.GetPChainAPI().BaseURL)
+	numConnected := 0
+	for _, peer := range peers {
+		vdrs, err := pClient.GetCurrentValidators(context.Background(), constants.PrimaryNetworkID, []ids.NodeID{peer.ID})
+		if err != nil {
+			panic(err)
+		}
+		// Only track the peer if it is a primary network validator
+		if len(vdrs) == 0 {
+			continue
+		}
+		logger.Info(
+			"Manually tracking bootstrap node",
+			zap.String("ID", peer.ID.String()),
+			zap.String("IP", peer.PublicIP.String()),
+		)
+		testNetwork.ManuallyTrack(peer.ID, peer.PublicIP)
+		numConnected++
+		if numConnected >= NumBootstrapNodes {
+			break
+		}
```

Comment:
Should we log something if we exit the `for` loop without connecting to `NumBootstrapNodes`?